src/main/java/com/cloudbees/AWSCredentialsUtils.java

```diff
@@ -0,0 +1,60 @@
+package com.cloudbees.jenkins.plugins.awscredentials;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.regions.DefaultAwsRegionProviderChain;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+
+import hudson.ProxyConfiguration;
+import jenkins.model.Jenkins;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class AWSCredentialsUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(AWSCredentialsUtils.class.getName());
+
+    private AWSCredentialsUtils() {}
+
+    static AWSSecurityTokenService buildStsClient(AWSCredentialsProvider provider) {
```

Comment:
So that we don't think it was forgotten in later maintenance

```suggestion
    /*package*/ static AWSSecurityTokenService buildStsClient(AWSCredentialsProvider provider) {
```